### ISSUE TITLE
Uptime CI Takes enourmously long to run and fails sometimes

### ISSUE THREAD

**Issue by asm0dey** [upptime/upptime#364]
**Describe the bug**
Uptime CI Takes enormously long to run and fails sometimes

**To Reproduce**

I have my own corn, which trigger workflow run every 5 minutes (because GitHub is not too reliable), and now there are lots of simultaneously running tasks

**Expected behavior**
The task should be executed faster than 5 minutes (at least for my 10 entries)

**Screenshots**
![image](https://user-images.githubusercontent.com/1297282/118636018-e5249480-b7dc-11eb-9c94-cae57b9e8346.png)

**Comment by AnandChowdhary** [comment]
Can you share a link to your repository? Super curious why it took 45 minutes.

**Comment by soemarko** [comment]
Having same problem...

<img width="937" alt="image" src="https://user-images.githubusercontent.com/381585/118669741-35711600-b820-11eb-876e-98a9524d4b1c.png">

**Comment by asm0dey** [comment]
@AnandChowdhary asm0dey/status

**Comment by AnandChowdhary** [comment]
I think this is related to GitHub Actions being down/unreliable right now: https://www.githubstatus.com/incidents/m16jzl31gnqt. Most likely this will fix itself as GitHub resolves this issue.

**Comment by asm0dey** [comment]
@AnandChowdhary task fail for several days already, but now it takes time:) 
Yesterday they were just momentarily failing :)

**Comment by asm0dey** [comment]
Now they all fail

**Comment by FxllenCode** [comment]
This is a severe issue for me currently. In the middle of an active incident it's currently taking forever to run the github actions:

![image](https://user-images.githubusercontent.com/40151425/119543980-44475380-bd5f-11eb-8ea2-e7804c416132.png)


This happened earlier too:

![image](https://user-images.githubusercontent.com/40151425/119544020-51fcd900-bd5f-11eb-8dab-1c5970087ba6.png)


Looking into this would be very appreciated!

(cc; @AnandChowdhary)

**Comment by FxllenCode** [comment]
Update: The issue is in the readme:

![Uc6I5k5s3p](https://user-images.githubusercontent.com/40151425/119544976-452cb500-bd60-11eb-8fed-a0dda2cd3448.gif)

**Comment by AnandChowdhary** [comment]
Aah, maybe it takes too long to fetch your (large) commit history. What if we stop it after like 1000 commits?

**Comment by asm0dey** [comment]
It actually looks like you need only 3*number of projects latest commits, right?

**Comment by AnandChowdhary** [comment]
We need commits to generate graphs, e.g., "1 year graph" of your uptime will need all commits for the past year. This of course doesn't scale well since we also offer an "all-time graph" so let me refactor this.

**Comment by FxllenCode** [comment]
> Aah, maybe it takes too long to fetch your (large,  100k+) commit history. What if we stop it after like 1000 commits?

1000 commits is more than enough, it might actually fix the big of graphs looking ridiculous at the all time scale. 

Thanks for the quick response!

**Comment by roim** [comment]
The generated png images are also fairly big. This repo is already well over 100MB in < 1 year.

Our page has more sites (so more graphs), and updates itself every minute. I think we'd exceed github repo size limits in little over a year. I suspect size is a bigger problem than # of commits, so would be nice to refactor graph rendering altogether--maybe not use pngs.

**Comment by AnandChowdhary** [comment]
> Would be nice to refactor graph rendering altogether--maybe not use pngs.

I agree, I was to phase them out entirely and instead rely upon dynamically generated SVGs.

**Comment by asm0dey** [comment]
Definitely +1 on replacing pngs with SVGs. Currently I think old images should be just purged with something like BFG

**Comment by asm0dey** [comment]
Definitely +1 on replacing pngs with SVGs. Currently I think old images should be just purged with something like BFG

**Comment by asm0dey** [comment]
Also maybe the script should not add new images, but just overwrite them?

**Comment by sid-r-singh** [comment]
> > Would be nice to refactor graph rendering altogether--maybe not use pngs.
> 
> I agree, I was to phase them out entirely and instead rely upon dynamically generated SVGs.

Why create PNGs at all ? Why not use the json data to create/embed graphs like [this](https://apexcharts.com/javascript-chart-demos/area-charts/datetime-x-axis/) one. I thought there will be heavy load on client for dealing with 1yr worth of data but here the graph is rendered & updated with ease (without any lag) even in mobile phones.

**Comment by FxllenCode** [comment]
Hey @AnandChowdhary, any update? This is a severe issue as I rely on the summary.json file for my tools and it does not update until way after because it takes so long to run summary ci... could this fix be expedited? Or is there something we can do to fix it while we wait?

**Comment by HugoGresse** [comment]
Our repo is over 8gb, this is preoccupying :/

**Comment by SukkaW** [comment]
> Our repo is over 8gb, this is preoccupying :/

My repo is at 1.5 GiB right now :/

**Comment by AnandChowdhary** [comment]
Hello friends, totally understand that it doesn't make sense to render images. I would love to accept a PR where we completely delete the graph generation and instead we can generate images on the fly using for example sparkcharts in the status page and README generation.
